### PR TITLE
Review changes done to publish package

### PR DESCRIPTION
**Pull request by AjazSumaiya**
## Description
This is a review branch for making the package public.

After review completes, this will be merged to main and a commit release will be pushed to the main branch that will 
publish the package.

## PR Related Changes
Changes done as part of this PR. 

-  removed the npmrc file from this public repo since it is not required anymore as we are publishing to open npm registry neither we will be using the package that is on artifactory as that has to be deprecated after this package gets released.
- updated package.json file to include the open github repo link, publish configs and version as 0.1.0.
- updated publish-npm.yml file taking in reference from the guild open-source formatter repo https://github.com/adobe/starter-repo/blob/master/.github/workflows/npm-publish.yml 
    > review the file https://github.com/adobe/plugin-hooks/blob/review/repo/.github/workflows/publish-to-npm.yml of this repo, if any changes need to be made.

- formatted repo referencing the open source template starter repo https://github.com/adobe/starter-repo

    > As part of formatting all these files were added License file in the root of the repo
    > - License headers in all code files
    > - Approved code of conduct
    > - Contribution guidelines

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by dthampy — APPROVED** [review]

File: .github/workflows/publish-to-npm.yml [hunk 1]
```diff
@@ -17,7 +17,10 @@ jobs:
...
-      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      NPM_AUTH_TOKEN: ${{ secrets.ADOBE_BOT_NPM_TOKEN }}
-      NPM_AUTH: ${{ secrets.ADOBE_BOT_NPM_TOKEN }}
+        with:
+          access: "public"
+          commit_pattern: "^Release (\\S+)"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_AUTH_TOKEN: ${{ secrets.ADOBE_BOT_NPM_TOKEN }}
+          NPM_AUTH: ${{ secrets.ADOBE_BOT_NPM_TOKEN }}
```
[the remark this answers is not in the document]
Comment (brasewel) [reply .github/workflows/publish-to-npm.yml:26]: I don't think these are required. If you look at how Deepty did it in the cli repo, you just need the adobe npm token. Please remove these and look at the cli for reference

**Review by brasewel — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-to-npm.yml [hunk 2]
```diff
@@ -17,7 +17,10 @@ jobs:
           node-version: 18
       - name: Publish if version has been updated
         uses: mkiki/npm-publish-action@c4315ef5790b7bcec2cbb75b34e37681a409d78d
-    env:
-      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      NPM_AUTH_TOKEN: ${{ secrets.ADOBE_BOT_NPM_TOKEN }}
-      NPM_AUTH: ${{ secrets.ADOBE_BOT_NPM_TOKEN }}
+        with:
+          access: "public"
+          commit_pattern: "^Release (\\S+)"
```
[the remark this answers is not in the document]
Comment (AjazSumaiya) [reply .github/workflows/publish-to-npm.yml:22]: Hi @amolina-adobe, yeah, the commit pattern you suggested seems better than this "^Release (\\S+)" as this is more restrictive match. 

This pattern i took from the starter repo template only https://github.com/adobe/starter-repo/blob/master/.github/workflows/npm-publish.yml. 

But you seem right it should match all the non-whitespace characters after the space.

File: .github/workflows/publish-to-npm.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (AjazSumaiya) [reply .github/workflows/publish-to-npm.yml:26]: yeah @amolina-adobe and @brasewel  i was not sure on this as the template seems to be updated from the one, we used in cli repo.

I replicated the template as is here.

But we are not using NPM_AUTH anywhere so we can remove it. maybe this naming is more conventional for .npmrc file that we are not using.

**Review by brasewel — APPROVED** [review]

File: .github/workflows/publish-to-npm.yml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 name: npm-publish
 on:
-  push:
```
Comment (brasewel) [inline .github/workflows/publish-to-npm.yml:3]: You will also want to follow how we do this in the cli where we only publish on the main branch

**Review by brasewel — CHANGES_REQUESTED** [review]

**Review by brasewel — CHANGES_REQUESTED** [review]
This change also needs to go into develop first

**Review by amolina-adobe — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-to-npm.yml [hunk 4]
```diff
@@ -17,7 +18,9 @@ jobs:
...
         uses: mkiki/npm-publish-action@c4315ef5790b7bcec2cbb75b34e37681a409d78d
-    env:
-      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      NPM_AUTH_TOKEN: ${{ secrets.ADOBE_BOT_NPM_TOKEN }}
-      NPM_AUTH: ${{ secrets.ADOBE_BOT_NPM_TOKEN }}
+        with:
+          access: "public"
+          commit_pattern: '^Release (\S+)'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
[the remark this answers is not in the document]
Comment (AjazSumaiya) [reply .github/workflows/publish-to-npm.yml]: removed the token as it is not required

**Review by amolina-adobe — APPROVED** [review]

**Review by brasewel — APPROVED** [review]

**Pull request merged**